types/function_list_builder.go

```diff
@@ -55,12 +56,19 @@ func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
 		if function.Annotations != nil {
 			annotations := *function.Annotations
 
-			if topic, pass := annotations["topic"]; pass {
+			if topicCSV, exists := annotations["topic"]; exists {
 
-				if serviceMap[topic] == nil {
-					serviceMap[topic] = []string{}
+				topicSlice := strings.Split(topicCSV, ",")
```

Comment:
Can you first check if the separator exists before attempting to Split? If it's not there then don't split. The separator needs to be configurable - if someone has a "," within a topic label then this would cause issues.

Reply:
Not sure `,` in a topic is valid
```
public static final String LEGAL_CHARS = "[a-zA-Z0-9._-]";
```

If it is, then the connector is already broken, isn't it?

https://github.com/openfaas-incubator/kafka-connector/blob/1c6be26bf1b524e8bc4fe135599dbf97e59b3fa2/main.go#L122-L128

Reply:
The connector-sdk is not only for Kafka, it's for any software which wishes to publish events.

Reply:
Of course.

Reply:
My comment was in reference to the Java regex you mentioned. Was it from the Kafka codebase?